source/idx/dffml_source_idx/idx1.py

```diff
@@ -8,16 +10,17 @@
 from dffml.repo import Repo
 from dffml.base import config
 from dffml.source.memory import MemorySource
-from dffml.source.file import FileSource, FileSourceConfig
+from dffml.source.file import (
+    FileSource,
+    FileSourceConfig,
+    BinaryFileSourceConfig,
+)
 from dffml.util.entrypoint import entrypoint
 
 
 @config
-class IDX1SourceConfig:
-    filename: str
-    feature: str
-    readwrite: bool = False
-    allowempty: bool = False
+class IDX1SourceConfig(BinaryFileSourceConfig):
+    feature: str = "unlabeled"
```

Comment:
I think we should probably leave this without a default argument. Also, if you change `from dffml.base import config` to `from dffml.base import config, field` then you can add a help string for it (see model/tensorflow for examples).

Reply:
`feature` can't be left without a default argument as it gives this error because the config class in file.py have a parameter without default argument :-
``TypeError: non-default argument 'feature' follows default argument``
So a default value is necessary here.

Reply:
Ahhh, that's because we're subclassing from `BinaryFileSourceConfig` we should make an issue to track this and make a fix for it eventually. For now could you just not subclass and just copy paste over the parameters from `BinaryFileSourceConfig`? Thanks! Then if we just cleanup the `docs/mnist.rst` file to only show the current training and accuracy commands, we can merge this and open a separate PR for the prediction stuff.